autopush/senderids.py

```diff
@@ -58,30 +59,32 @@ def __init__(self, args):
                 log.err("senderid_list is not a dict. Ignoring")
             else:
                 self.update(senderIDs)
-        self._refresh()
+        # set up auto-refresh for senderIDs
+        if self._use_s3:
+            self.service = TimerService(self._expry, self._refresh)
+            self.service.startService()
 
+    @inlineCallbacks
     def _write(self, bucket, senderIDs):
         """Write a list of SenderIDs to S3"""
         key = Key(bucket)
         key.key = self.KEYNAME
-        key.set_contents_from_string(json.dumps(senderIDs))
+        yield key.set_contents_from_string(json.dumps(senderIDs))
```

Comment:
This method returns a deferred? inlineCallbacks using yield should be yielding deferreds.